### ISSUE TITLE
loss curve

### ISSUE THREAD

**Issue by pokameng** [M-E-AGI-Lab/Muddit#5]
@viiika 
Can you provide the loss curve?
I have some questions about this loss

**Comment by pokameng** [comment]
1.I train COCO dataset, but the loss ratio is between 8~9,it seems not good.
2.I inference on resolution of 1024*1024 on your codebase, it can generate image, but not generating image on 256*256.

**Comment by viiika** [comment]
Thank you for your interest!

In our experience, the training loss curve can vary significantly with different datasets. Changing the training data often results in abrupt shifts in the loss values, making it uninformative to share our specific training curve. (See also https://github.com/viiika/Meissonic/issues/25 for related discussion.)

Regarding your two questions:
1. A training loss in the range of 7–9 on the COCO dataset is normal. Could you clarify which aspect you find not good? If you are training with 1024 resolution, we recommend setting the text loss weight to 0.3. In this case, the loss typically starts around 9 and converges to about 7–8.
2. When you say “not generating”, do you mean a runtime error (the model fails to run), or that the generated outputs have poor quality?

**Comment by pokameng** [comment]
@viiika 
I also inference based on codebase :https://github.com/viiika/Meissonic/tree/main
1024*1024：
![Image](https://github.com/user-attachments/assets/ddc5dc47-c266-4642-ad10-40426f5fa7b4)
512*512：
<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/35740aa5-7efa-424f-ad7e-7378d8f37127" />
256*256：

<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/464e50d6-3414-4237-bd7a-47e74ed8681e" />

The image with resolution of 1024*1024 seems good, but 512*512 and 256*256 are worse.
Can you give me some guidances?
Thanks!!!

**Comment by pokameng** [comment]
@viiika 
If we use 256*256 COCO, is it appropriate to train muddit from scratch? I want to verify whether muddit supports 256*256

**Comment by pokameng** [comment]
@viiika 
Are the weights of muddit in 512*512 fine-tuned using the weights of codebase https://github.com/viiika/Meissonic/tree/main? based on resolution 1024*1024?

I have a lot of questions, sorry to bother you.
Thanks！！！

**Comment by pokameng** [comment]
@viiika 

<img width="690" height="547" alt="Image" src="https://github.com/user-attachments/assets/547a50a9-5759-49ba-9d6d-b4cd60587d7f" />
In training step, the  timestep *1000, and in the SymmetricTransformer2DModel-inner, the timestep also timestep *1000，this very strange.

<img width="568" height="152" alt="Image" src="https://github.com/user-attachments/assets/f066b8be-f299-4a9a-89b4-36d73ae69d54" />

In print the timestep,before and after,:
example:
before:[998]
after:[inf]

**Comment by viiika** [comment]
Sorry for the late reply, as I have been traveling this week.

I will check the details with co-authors and get back to you as soon as possible.

If you have further questions, please let me know. We will try our best to assist your research.

**Comment by viiika** [comment]
Sorry for the late reply again, I'm back!

Q1: The image quality at 1024×1024 looks good, but 512×512 and 256×256 are worse.
A1: That’s expected due to current limitations in Meissonic. The model was not trained with a dynamic-resolution strategy, so it performs best at 1024×1024. If you're interested in enabling multi-resolution support, we’d be happy to share more detailed suggestions. You can also refer to the bucket description of section 5.3.2 and E.1 from https://arxiv.org/pdf/2403.03206.

Q2: Is it appropriate to train Muddit from scratch on 256×256 COCO? I want to verify if Muddit supports this resolution.
A2: Yes, it does, feel free to try it. The key part to check is  https://github.com/M-E-AGI-Lab/Muddit/blob/12d9ed20fd767379ca37608e4a1581f4c2a8e3be/src/pipeline.py#L536C14-L551C22. For reference, in the original Meissonic implementation, we doubled the size of img_ids when running under 512 conditions (https://github.com/viiika/Meissonic/blob/8e590ee4cff5c10669816660106ccc11bf93a374/src/pipeline.py#L306). If you keep the same downblock/upblock structure, you can leave it unchanged; otherwise, remember to double it accordingly.

Q3: Are the 512×512 weights of Muddit fine-tuned from Meissonic (1024×1024)?
A3: Yes. Both Muddit (512) and Muddit Plus (1024) are initialized from Meissonic (1024), using its downsampling and upsampling blocks.

Q4: Before: [998], After: [inf]
A4: Try switching the data type from fp16 to bf16. This issue stems from a design shortcoming in Meissonic. We attempted to remove one *1000 before, but that led to a slight drop in performance.

Hope these clarify your questions!
If you have further issues, feel free to ask here. For faster communication, you can also reach me via WeChat or email.

**Comment by pokameng** [comment]
> Sorry for the late reply again, I'm back!
> 
> Q1: The image quality at 1024×1024 looks good, but 512×512 and 256×256 are worse. A1: That’s expected due to current limitations in Meissonic. The model was not trained with a dynamic-resolution strategy, so it performs best at 1024×1024. If you're interested in enabling multi-resolution support, we’d be happy to share more detailed suggestions. You can also refer to the bucket description of section 5.3.2 and E.1 from https://arxiv.org/pdf/2403.03206.
> 
> Q2: Is it appropriate to train Muddit from scratch on 256×256 COCO? I want to verify if Muddit supports this resolution. A2: Yes, it does, feel free to try it. The key part to check is https://github.com/M-E-AGI-Lab/Muddit/blob/12d9ed20fd767379ca37608e4a1581f4c2a8e3be/src/pipeline.py#L536C14-L551C22. For reference, in the original Meissonic implementation, we doubled the size of img_ids when running under 512 conditions (https://github.com/viiika/Meissonic/blob/8e590ee4cff5c10669816660106ccc11bf93a374/src/pipeline.py#L306). If you keep the same downblock/upblock structure, you can leave it unchanged; otherwise, remember to double it accordingly.
> 
> Q3: Are the 512×512 weights of Muddit fine-tuned from Meissonic (1024×1024)? A3: Yes. Both Muddit (512) and Muddit Plus (1024) are initialized from Meissonic (1024), using its downsampling and upsampling blocks.
> 
> Q4: Before: [998], After: [inf] A4: Try switching the data type from fp16 to bf16. This issue stems from a design shortcoming in Meissonic. We attempted to remove one *1000 before, but that led to a slight drop in performance.
> 
> Hope these clarify your questions! If you have further issues, feel free to ask here. For faster communication, you can also reach me via WeChat or email.

maybe we can chat with Wechat:
IMAG-WSM

**Comment by pokameng** [comment]
> Sorry for the late reply again, I'm back!
> 
> Q1: The image quality at 1024×1024 looks good, but 512×512 and 256×256 are worse. A1: That’s expected due to current limitations in Meissonic. The model was not trained with a dynamic-resolution strategy, so it performs best at 1024×1024. If you're interested in enabling multi-resolution support, we’d be happy to share more detailed suggestions. You can also refer to the bucket description of section 5.3.2 and E.1 from https://arxiv.org/pdf/2403.03206.
> 
> Q2: Is it appropriate to train Muddit from scratch on 256×256 COCO? I want to verify if Muddit supports this resolution. A2: Yes, it does, feel free to try it. The key part to check is https://github.com/M-E-AGI-Lab/Muddit/blob/12d9ed20fd767379ca37608e4a1581f4c2a8e3be/src/pipeline.py#L536C14-L551C22. For reference, in the original Meissonic implementation, we doubled the size of img_ids when running under 512 conditions (https://github.com/viiika/Meissonic/blob/8e590ee4cff5c10669816660106ccc11bf93a374/src/pipeline.py#L306). If you keep the same downblock/upblock structure, you can leave it unchanged; otherwise, remember to double it accordingly.
> 
> Q3: Are the 512×512 weights of Muddit fine-tuned from Meissonic (1024×1024)? A3: Yes. Both Muddit (512) and Muddit Plus (1024) are initialized from Meissonic (1024), using its downsampling and upsampling blocks.
> 
> Q4: Before: [998], After: [inf] A4: Try switching the data type from fp16 to bf16. This issue stems from a design shortcoming in Meissonic. We attempted to remove one *1000 before, but that led to a slight drop in performance.
> 
> Hope these clarify your questions! If you have further issues, feel free to ask here. For faster communication, you can also reach me via WeChat or email.

Can you tell me your Wechat?

**Comment by viiika** [comment]
Since the inference pipeline only multiplies by 1000 once, we can simply modify the code to remove the redundant *1000.

Thanks for pointing out this important issue!
